benchmark/benchmark.py

```diff
@@ -149,3 +149,20 @@ def test_get_value_kind(
 
 def test_is_missing_literal(benchmark: Any) -> None:
     assert benchmark(_is_missing_literal, "???")
+
+
+@mark.parametrize("force_add", [False, True])
+def test_update_force_add(
+    large_dict_config: Any, force_add: bool, benchmark: Any
+) -> None:
```

Comment:
Added a benchmark to verify the excessive flag overrides are not causing performance regression.
The concern is that the recursive flags cache invalidation when a flag is set will slow things down.
It does not seem to cause a significant slowdown for a large dict_config here though.

Reply:
Yeah, I didn't notice the gating in _invalidate_flags_cache.
Good point about the benchmark not doing much.

I fixed the benchmark to populate the flags cache on all nodes. We don't have a baseline though.
Overall I am not too concerned about relative performance here though, at 400us per set on the big config this is plenty fast enough.